src/view/ThemeSettings.js

```diff
@@ -100,7 +100,24 @@ define(function (require, exports, module) {
                 var attr = $target.attr("data-target");
                 newSettings[attr] = $target.is(":checked");
             })
-            .on("input", "[data-target]:text", function () {
+            .on("input", "#font-size-input", function () {
+                var $target = $(this);
+                var $input = $target.val();
+                var validInput = /^[\d\.]+(p|px|e|em){0,1}$/g;
+                var btn = document.getElementById("done");
+
+                // Make sure that the font size is expressed in terms we can handle (px or em). If not, 'done' button disabled until input corrected.
+
+                if ($input.search(validInput) === -1) {
+                    ($input.length === 0) ? btn.disabled=false : btn.disabled=true;
+                    return false;
+                }else{
```

Comment:
`}else{` after `return` doesn't make sense, just close the if with `}` and continue normally